include/algorithms/decision_forest/decision_forest_classification_predict.h

```diff
@@ -305,8 +305,148 @@ class Batch : public classifier::prediction::Batch
+ *        and supports method to compute decision_forest prediction
+ *
+ * \tparam algorithmFPType  Data type to use in intermediate computations for the decision_forest, double or float
+ * \tparam method           decision_forest computation method, \ref Method
+ */
+template <typename algorithmFPType, Method method, CpuType cpu>
+class BatchContainer : public PredictionContainerIface
+{
+public:
+    /**
+     * Constructs a container for decision_forest model-based prediction with a specified environment
+     * \param[in] daalEnv   Environment object
+     */
+    BatchContainer(daal::services::Environment::env * daalEnv);
+    /** Default destructor */
+    ~BatchContainer();
+    /**
+     * Computes the result of decision_forest model-based prediction
+     * \return Status of computations
+     */
+    services::Status compute() DAAL_C11_OVERRIDE;
+};
+
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__DECISION_FOREST__CLASSIFICATION__PREDICTION__BATCH"></a>
+ * \brief Predicts decision_forest classification results
+ * <!-- \n<a href="DAAL-REF-decision_forest-ALGORITHM">decision_forest algorithm description and usage models</a> -->
+ *
+ * \tparam algorithmFPType  Data type to use in intermediate computations for the decision_forest algortithm, double or float
+ * \tparam method           decision_forest computation method, \ref Method
+ *
+ * \par Enumerations
+ *      - \ref Method                                       decision_forest prediction methods
+ *      - \ref classifier::prediction::NumericTableInputId  Identifiers of input Numeric Table objects
+ *                                                          for the decision_forest prediction algorithm
+ *      - \ref classifier::prediction::ModelInputId         Identifiers of input Model objects of the decision_forest prediction algorithm
+ *      - \ref classifier::prediction::ResultId             Identifiers of decision_forest prediction results
+ *
+ * \par References
+ *      - \ref interface1::Model "Model" class
+ *      - \ref classifier::prediction::interface1::Input "classifier::prediction::Input" class
+ *      - \ref classifier::prediction::interface1::Result "classifier::prediction::Result" class
+ */
+template <typename algorithmFPType = DAAL_ALGORITHM_FP_TYPE, Method method = defaultDense>
+class Batch : public classifier::prediction::Batch
+{
+public:
+    typedef classifier::prediction::Batch super;
+
+    typedef algorithms::decision_forest::classification::prediction::Input InputType;
+    typedef algorithms::decision_forest::classification::prediction::Parameter ParameterType;
+    typedef typename super::ResultType ResultType;
+
+    InputType input;         /*!< %Input objects of the algorithm */
+
+    /**
+     * Constructs Decision forest prediction algorithm
+     * \param[in] nClasses  Number of classes
+     */
+    Batch(size_t nClasses)
+    {
+        _par = new ParameterType(nClasses);
```

Comment:
it should be hidden in kernel, otherwise there is no effect of parameter() implementation!
Object of Parameter class should be constructed on library side.